### PR TITLE
Make a slash command to get information of all the contributors #38

### PR DESCRIPTION
**Pull request by Sarthak5598**
OWASP-BLT/BLT-Sammich#1

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -216,3 +218,59 @@ def list_project():
...
+    for comment in comments:
+        user = comment['user']['login']
+        if user not in user_data:
+            user_data[user] = {'prs': 0, 'issues': 0, 'comments': 0}
+        user_data[user]['comments'] += 1
+
+    table = "User | PRs Merged | Issues Resolved | Comments\n"
+    table += "---- | ---------- | --------------- | --------\n"
+    for user, counts in user_data.items():
+        table += f"{user} | {counts['prs']} | {counts['issues']} | {counts['comments']}\n"
```
Comment (arkid15r) [inline app.py:253]: Please don't use `+=` for strings as it's quite inefficient. You can use a list and then join items with `\n`

File: app.py [hunk 2]
```diff
@@ -3,16 +3,18 @@
...
 
 import json
+import requests
 import git
 from dotenv import load_dotenv
 from flask import Flask, jsonify, request
 from slack import WebClient
 from slack_sdk.errors import SlackApiError
 from slackeventsapi import SlackEventAdapter
+from datetime import datetime, timedelta
```
Comment (arkid15r) [inline app.py:13]: These are unused.

File: app.py [hunk 3]
```diff
@@ -216,3 +218,59 @@ def list_project():
...
+        table += f"{user} | {counts['prs']} | {counts['issues']} | {counts['comments']}\n"
+    
+    return table
+
+@app.route('/contributors', methods=['POST'])
+def contributors():
+    data = request.form
+    owner = "OWASP-BLT"
+    repo = "BLT"
+    slack_channel = data.get('channel_id')
```
Comment (arkid15r) [inline app.py:262]:
```suggestion
```

**Review by arkid15r — CHANGES_REQUESTED** [review]
It's a cool functionality extension, thanks @Sarthak5598!

Could you take a look at addressing the following when you get a chance:

File: app.py [hunk 4]
```diff
@@ -216,3 +218,59 @@ def list_project():
             "text": message,
         }
     )
+
+def fetch_github_data(owner, repo):
+    prs = requests.get(f'{GITHUB_API_URL}/repos/{owner}/{repo}/pulls?state=closed').json()
+    issues = requests.get(f'{GITHUB_API_URL}/repos/{owner}/{repo}/issues?state=closed').json()
+    comments = requests.get(f'{GITHUB_API_URL}/repos/{owner}/{repo}/issues/comments').json()
```
[the remark this answers is not in the document]
Comment (Sarthak5598) [reply app.py:225]: Thank you for the reccomendation ! , Can you please tell me what is the access token ?

File: app.py (hunk 1 above)
Comment (Sarthak5598) [reply app.py:253]: sure!

File: app.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (arkid15r) [reply app.py:224]: Here is an official [doc](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28). Generally, official documentation has/describes all terms.

**Comment by Sarthak5598** [comment]
Thanks for all the recommendations @arkid15r !

File: app.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Sarthak5598) [reply app.py:224]: Thank you !

File: app.py [hunk 5]
```diff
@@ -219,3 +220,52 @@ def list_project():
...
+
+    for comment in comments:
+        user = comment['user']['login']
+        if user not in user_data:
+            user_data[user] = {'prs': 0, 'issues': 0, 'comments': 0}
+        user_data[user]['comments'] += 1
+
+    table = "User | PRs Merged | Issues Resolved | Comments\n ---- | ---------- | --------------- | --------\n"
+    for user, counts in user_data.items():
+        table += f"{user} | {counts['prs']} | {counts['issues']} | {counts['comments']}\n"
```
Comment (arkid15r) [inline app.py:254]: I suggest using list and append rows to it. Then you can join it with `\n`. Don't use string concatenation.

**Review by arkid15r — COMMENTED** [review]
Please address the `pre-commit` errors

**Pull request closed without merging**